src/components/Web3Status/index.tsx

```diff
@@ -146,6 +146,15 @@ function Web3StatusInner() {
 
   const { hasPendingActivity, pendingActivityCount } = usePendingActivity()
 
+  const displayName = useMemo(() => (account ? ENSName || shortenAddress(account) : null), [account, ENSName])
+  const lastDisplayName = useRef(localStorage.getItem(selectedWalletDisplayKey) ?? null)
+  useEffect(() => {
+    if (displayName) {
+      // It takes a moment to load the ENSName, so set the lastDisplayName (for next time) to the account to reduce layout shift.
+      localStorage.setItem(selectedWalletDisplayKey, shortenAddress(account))
```

Comment:
why is lastDisplayName set to `shortenAddress(account)` and not `displayName`?

Reply:
Because the account address will load before ens. We could tighten this up more but this was a first attempt. 

See inline comment above this line.